### PR TITLE
QoL fixes

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Framework/Graphics/Drawable.cs [hunk 1]
```diff
@@ -22,6 +22,11 @@ public abstract partial class Drawable : IDisposable, IHasLifetime
 
         internal event Action OnInvalidate;
 
+        /// <summary>
+        /// A name used to identify this Drawable internally.
+        /// </summary>
+        public virtual string Name => null;
```
Comment (smoogipooo) [inline osu.Framework/Graphics/Drawable.cs]: would prefer string.Empty instead of null.

File: osu.Framework/Graphics/Transformations/Transform.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ public abstract class Transform<T> : ITransform
 
         protected double Time => Clock.CurrentTime;
 
-        public bool IsAlive => Clock == null || (StartTime <= Clock.CurrentTime && EndTime >= Clock.CurrentTime);
+        public bool IsAlive => Clock == null || (StartTime <= Clock.CurrentTime && (EndTime >= Clock.CurrentTime || LoopCount != CurrentLoopCount));
```
Comment (SirCmpwn) [inline osu.Framework/Graphics/Transformations/Transform.cs]: At this point this probably shouldn't be a shorthand property anymore.

File: osu.Framework/Graphics/Transformations/Transform.cs [hunk 3]
```diff
@@ -26,7 +26,21 @@ public abstract class Transform<T> : ITransform
...
         protected double Time => Clock.CurrentTime;
 
-        public bool IsAlive => Clock == null || (StartTime <= Clock.CurrentTime && EndTime >= Clock.CurrentTime);
+        public bool IsAlive
+        {
+            get
+            {
+                if (Clock == null)
+                    //we could not be completely initialised yet.
+                    return true;
```
Comment (SirCmpwn) [inline osu.Framework/Graphics/Transformations/Transform.cs]: Style nit - I don't like putting comments or anything at all between the if statement and the body if braces are omitted. I'd join the comment onto the return line or put it above the if statement, personally.

Comment (peppy) [inline osu.Framework/Graphics/Transformations/Transform.cs]: hmm ok. i feel like if i put it before the if statement (the other place i'd put it) i need to reword it to be pre-conditional

Comment (SirCmpwn) [inline osu.Framework/Graphics/Transformations/Transform.cs]: Or after the ) also works without rewording

**Comment by SirCmpwn** [comment]
:+1:
